src/formatDate.js

```diff
@@ -8,7 +8,26 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+  const dateParts = date.split(fromSeparator);
+  const dateMap = fromFormat.reduce((map, part, index) => {
+    map[part] = dateParts[index];
+
+    return map;
+  }, {});
+
+  if (dateMap['YYYY'] && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
```

Comment:
The conversion from 'YYYY' to 'YY' is correct, but make sure to also delete the 'YYYY' key from `dateMap` after the conversion to avoid potential issues when building `newDateParts`.